geomesa-features/geomesa-feature-common/src/main/scala/org/locationtech/geomesa/features/ScalaSimpleFeature.scala

```diff
@@ -26,19 +26,23 @@ import scala.collection.JavaConversions._
 /**
  * Simple feature implementation optimized to instantiate from serialization
  *
- * @param initialId
- * @param sft
+ * @param initialId simple feature id
+ * @param sft simple feature type
  * @param initialValues if provided, must already be converted into the appropriate types
  */
-class ScalaSimpleFeature(initialId: String, sft: SimpleFeatureType, initialValues: Array[AnyRef] = null)
+class ScalaSimpleFeature(initialId: String,
+                         sft: SimpleFeatureType,
+                         initialValues: Array[AnyRef] = null,
+                         initialUserData: java.util.Map[AnyRef, AnyRef] = null)
```

Comment:
I didn't want to use the input object directly (as it's a mutable object), so using null avoids allocating empty maps when we don't have to.